packages/gatsby/src/utils/show-experiment-notice.ts

```diff
@@ -35,3 +56,31 @@ export function showExperimentNoticeAfterTimeout(
     clearTimeout(noticeTimeout)
   }
 }
+
+export const createNoticeMessage = (notices): string => {
+  let message = `\nHi from the Gatsby maintainers! Based on what we see in your site, these coming
+features may help you. All of these can be enabled within gatsby-config.js via
+flags (samples below)`
+
+  notices.forEach(
+    notice =>
+      (message += `
+
+${chalk.bgBlue.bold(
+  terminalLink(notice.experimentIdentifier, notice.umbrellaLink)
+)}, ${notice.noticeText}\n`)
+  )
+
+  return message
+}
+
+const showNotices = (): void => {
+  emitter.off(`COMPILATION_DONE`, showNotices)
+  if (noticesToShow.length > 0) {
+    telemetry.trackFeatureIsUsed(`InviteToTryExperiment`)
+    const message = createNoticeMessage(noticesToShow)
+    reporter.info(message)
+  }
+}
+
+emitter.on(`COMPILATION_DONE`, showNotices)
```

Comment:
Can we avoid executing code at import time? This leads to weird and hard to debug problems later on when shuffling imports that _potentially_ impact how things work together. In fact I would propose to not even use `COMPILATION_DONE` callback at all and just call `showNotices()` in webpack's DONE hook handler (on first compile only?)

Reply:
yeah we could do that 👍